### PR TITLE
RUBY-2376 Query cache documentation

### PR DESCRIPTION
**Pull request by egiurleo**
I have moved the query cache documentation to its own page and linked to it from the CRUD page. I have gone in depth on various topics, such as cache invalidation, using the query cache with transactions and aggregations, and queries with limits.

### CODE REVIEW

File: docs/tutorials/query-cache.txt [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+    client['artists'].find(genre: 'Rock', limit: 20)
+    #=> Queries the database again and replaces the previously cached query results
+  end
+
+Cache Invalidation
+==================
+
+The query cache is cleared in part or in full on every write operation. Most
+write operations will clear any queries were performed on the same
```
Comment (p-mongo) [inline docs/tutorials/query-cache.txt:94]: Suggest "will clear results of any queries that were performed"

File: docs/tutorials/query-cache.txt [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+collection that is being written to. Some operations will clear the entire
+query cache.
+
+The following operations will clear cached queries on the same database and
```
Comment (p-mongo) [inline docs/tutorials/query-cache.txt:98]: Suggest "cached query results"

**Review by p-mongo — APPROVED** [review]
This is excellent!

Could you also add a section that talks about the scope of each cached result, to specifically address the lack of cache clearing when transaction is starting (i.e. so that users understand that if they start a tx with snapshot read concern, the existing entries won't be used, and if continue with the previous write concern they may end up reading cache entries in a transaction).

File: docs/tutorials/query-cache.txt [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. _query-cache:
+
+The MongoDB Ruby driver provides a built-in query cache. When enabled, the
+query cache saves the results of previously-executed find and aggregation
+queries. When those same queries are performed again, the driver returns
+the cached reuslts to prevent unnecessary round trips to the database.
```
[the remark this answers is not in the document]
Comment (egiurleo) [reply docs/tutorials/query-cache.txt:18]: You're right, it's "roundtrip."

File: docs/tutorials/query-cache.txt [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+* Skip
+* Sort
+* Projection
+* Collation
+* Read Concern
+* Read Preference
+
+For example, if you perform one query, and then perform a mostly identical query
+with a different sort order, those queries will not be considered matching,
+and the second query will not use the cached results of the first.
```
Comment (egiurleo) [inline docs/tutorials/query-cache.txt:89]: @p-mongo What do you think of this section?

Comment (p-mongo) [reply docs/tutorials/query-cache.txt:89]: :+1:

**Review by p-mongo — APPROVED** [review]

**Pull request merged**